src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/Everything/PatientEverythingService.cs

```diff
@@ -177,6 +238,157 @@ public class PatientEverythingService : IPatientEverythingService
+
+                // Regardless of the other links present, we throw an error if we find a "replaced-by" link.
+                if (link.Type == Patient.LinkType.ReplacedBy)
+                {
+                    // Specify the url that must be used instead.
+                    Uri url = _urlResolver.ResolveOperationResultUrl(OperationsConstants.PatientEverything, referenceSearchValue.ResourceId);
+
+                    throw new EverythingOperationException(
+                        string.Format(
+                            Core.Resources.EverythingOperationResourceIrrelevant,
+                            parentPatientId,
+                            referenceSearchValue.ResourceId),
+                        HttpStatusCode.MovedPermanently,
+                        url.ToString());
+                }
+            }
+        }
+
+        private List<Patient.LinkComponent> ExtractLinksFromParentPatient(SearchResultEntry parentPatientResource)
+        {
+            // If it wasn't found, it means the parent patient doesn't exist in the database.
+            if (parentPatientResource.Resource == null)
+            {
+                // There are no links to extract since there is no parent patient resource.
+                return null;
+            }
+
+            // Otherwise, we found the parent patient in the database. Extract its links.
+            ResourceElement element = _resourceDeserializer.Deserialize(parentPatientResource.Resource);
+            Patient parentPatient = element.ToPoco<Patient>();
+
+            return parentPatient.Link;
+        }
+
+        private async Task<string> CheckForNextSeeAlsoLinkAndSetToken(string parentPatientId, EverythingOperationContinuationToken token, CancellationToken cancellationToken)
+        {
+            // Retrieve the parent patient so that we can extract its links and process the next "seealso" link.
+            using IScoped<ISearchService> search = _searchServiceFactory();
+            var searchResultEntries = new List<SearchResultEntry>();
+
+            // To do so, first create the search parameter to add to the query
+            var searchParameters = new List<Tuple<string, string>>
+            {
+                Tuple.Create(SearchParameterNames.Id, parentPatientId),
+            };
+
+            // And then execute the search.
+            SearchOptions searchOptions = _searchOptionsFactory.Create(ResourceType.Patient.ToString(), searchParameters);
+            SearchResult searchResult = await search.Value.SearchAsync(searchOptions, cancellationToken);
+            searchResultEntries.AddRange(searchResult.Results.Select(x => new SearchResultEntry(x.Resource)));
+
+            // The search result entries should contain one patient resource - the parent patient.
+            SearchResultEntry parentPatientResource = searchResultEntries.FirstOrDefault(s => string.Equals(s.Resource.ResourceTypeName, ResourceType.Patient.ToString(), StringComparison.Ordinal));
```

Comment:
Do we have a special reason to create `searchResultEntries` here? It is created in `SearchIncludes` to change the search mode from "include" to "match", but it looks not needed in this function. We can consider using `searchResult.Results.FirstOrDefault()` directly.

Reply:
Fixed in 3188b31.